pkg/catalog/routes.go

```diff
@@ -460,3 +460,48 @@ func (mc *MeshCatalog) GetServicesForServiceAccounts(saList []service.K8sService
 
 	return serviceList
 }
+
+// GetHostnamesForUpstreamService returns the hostnames over which an upstream service is accessible from a downstream service
+// The hostname is the FQDN for the service, and can include ports as well.
+// Ex. bookstore.default, bookstore.default:80, bookstore.default.svc, bookstore.default.svc:80 etc.
+// TODO: replace GetResolvableHostnamesForUpstreamService with this func once routes refactor is complete (#issue)
+func (mc *MeshCatalog) GetHostnamesForUpstreamService(downstream, upstream service.MeshService) ([]string, error) {
+	sameNamespace := downstream.Namespace == upstream.Namespace
+	// The hostnames for this service are the Kubernetes service DNS names
+	hostnames, err := mc.getServiceHostnames(upstream, sameNamespace)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error getting service hostnames for upstream service %s", upstream)
+		return nil, err
+	}
+
+	return hostnames, nil
+}
+
+// buildTrafficPolicies takes a list of source services, destination services and routes and returns a list of traffic policies for all
+//	combinations of the given source and destination services
+func (mc *MeshCatalog) buildTrafficPolicies(sourceServices, destServices []service.MeshService, routes []trafficpolicy.HTTPRoute) (policies []*trafficpolicy.TrafficPolicy) {
+	for _, sourceService := range sourceServices {
+		for _, destService := range destServices {
+			if sourceService == destService {
+				continue
+			}
+			routesClusters := []trafficpolicy.RouteWeightedClusters{}
+			weightedClusters := mapset.NewSet(getDefaultWeightedClusterForService(destService))
```

Comment:
Maybe we need a comment here for why we are not getting the traffic split backends here

Reply:
added

Reply:
added